src/java/picard/illumina/IlluminaBasecallsToFastq.java

```diff
@@ -68,15 +68,32 @@
 import java.util.Set;
 
 @CommandLineProgramProperties(
-        usage = "Generate fastq file(s) from data in an Illumina basecalls output directory.\n" +
-                "Separate fastq file(s) are created for each template read, and for each barcode read, in the basecalls.\n" +
-                "Template fastqs have extensions like .<number>.fastq, where <number> is the number of the template read,\n" +
-                "starting with 1.  Barcode fastqs have extensions like .barcode_<number>.fastq, where <number> is the number\n" +
-                "of the barcode read, starting with 1.",
-        usageShort = "Generate fastq file(s) from data in an Illumina basecalls output directory",
-        programGroup = Illumina.class
+  usage = IlluminaBasecallsToFastq.USAGE_SUMMARY + IlluminaBasecallsToFastq.USAGE_DETAILS,
+  usageShort = IlluminaBasecallsToFastq.USAGE_SUMMARY,
+  programGroup = Illumina.class
 )
 public class IlluminaBasecallsToFastq extends CommandLineProgram {
+  static final String USAGE_SUMMARY = "Generate fastq file(s) from Illumina basecall read data.  ";
+  static final String USAGE_DETAILS = "<p>This tool generates fastq files from data in an Illumina basecalls (*.bcl) output directory.  " +
+          "Separate fastq files are created for each template, barcode, and index read.   In the absence of sample pooling (multiplexing) and/or barcodes, then an OUTPUT_PREFIX (file) must be provided as the sample identifier.  For multiplexed samples, a MULTIPLEX_PARAMS (file) must be used.  The MULTIPLEX_PARAMS file contains the list of sample barcodes used to sort template, barcode, and index reads.  It is essentially the same as the BARCODE_FILE used in the" +
```

Comment:
exactly. we use it all the time.

On Wed, May 25, 2016 at 10:10 AM, Geraldine Van der Auwera <
notifications@github.com> wrote:

> In src/java/picard/illumina/IlluminaBasecallsToFastq.java
> <https://github.com/broadinstitute/picard/pull/521#discussion_r64579007>:
>
> >  )
> >  public class IlluminaBasecallsToFastq extends CommandLineProgram {
> > +  static final String USAGE_SUMMARY = "Generate fastq file(s) from Illumina basecall read data.  ";
> > +  static final String USAGE_DETAILS = "<p>This tool generates fastq files from data in an Illumina basecalls (*.bcl) output directory.  " +
> > +          "Separate fastq files are created for each template, barcode, and index read.   In the absence of sample pooling (multiplexing) and/or barcodes, then an OUTPUT_PREFIX (file) must be provided as the sample identifier.  For multiplexed samples, a MULTIPLEX_PARAMS (file) must be used.  The MULTIPLEX_PARAMS file contains the list of sample barcodes used to sort template, barcode, and index reads.  It is essentially the same as the BARCODE_FILE used in the" +
>
> @yfarjoun <https://github.com/yfarjoun> do we really expect the output
> prefix to be a path? That doesn't make a whole lot of sense to me. Is that
> for when we want the output to go someplace that isn't the working
> directory?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/picard/pull/521/files/eb30e0911b5269b06a336470afc1d258b54283e2#r64579007>
>